src/libexec/zmsearchndelete

```diff
@@ -0,0 +1,941 @@
+                next;
+            }
+
+            my $rc = _chk_acct( $smtp, $ahash, \%exclude );
+            $exclcnt++ if ( !defined($rc) );
+            push( @accts, $smtp ) if ($rc);
+        }
+    }
+
+    warn("$Prog: note: $exclcnt account(s) excluded via exclude list(s)\n")
+      if ($exclcnt);
+
+    return \@accts;
+}
+
+# error handling in call_soap sub
+sub _err {
+    my ($resp) = @_;
+    Carp::confess("$Prog: no response object\n") unless ($resp);
+    return ( $resp->valueof('//Fault/Reason/Text') || "<unknown>" )
+      . ( $DEBUG > 1 ? ( ": " . Dumper( $resp->fault ) ) : "" );
+}
+
+# format: 02 Oct 2005 07:50:32 -0500 (EST)
+sub _ts {
+    return POSIX::strftime( "%d %b %Y %H:%M:%S %z (%Z)", localtime(time) );
+}
+
+sub _date {
+    my $dt = $_[0];
+    return POSIX::strftime( "%D %T", localtime($dt) );
+}
+
+__END__
+
+=head1 Input File Format
+
+The list of users to update can be read from the command line or via
+input files.  For the input file, blank lines, lines with only white
+space, and lines beginning with (optional whitespace) and then '#' are
+all skipped.
+
+Entries in the input file are simply in the format of one "user" per
+line, where user is in the format of "user@domain".
+
+Examples:
+
+  user1@dom1.name
+  user2@dom2.name
+  user3@dom3.name
+
+=head1 Performance
+
+Searching serially through mailboxes is likely to be slow.  Some
+attempts have been made to tune this utility to reduce run time where
+possible.  This section describes some of the tuning performed:
+
+=head2 SearchRequest limit
+
+The ZCS SearchRequest SOAP call provides paged results.  Below are
+timings with various values for 'limit'.
+
+Test mailbox with 16674 msgs.  A limit of 200 was chosen after these
+tests.
+
+ limit: timing via Benchmark
+ ------ --------------------
+   250: 238 wallclock secs (151.49 usr +  0.12 sys = 151.61 CPU)
+   225: 233 wallclock secs (137.40 usr +  0.10 sys = 137.50 CPU)
+   200: 227 wallclock secs (124.16 usr +  0.11 sys = 124.27 CPU)
+   150: 234 wallclock secs (98.25 usr +  0.13 sys = 98.38 CPU)
+   100: 282 wallclock secs (71.88 usr +  0.15 sys = 72.03 CPU)
+    75: 318 wallclock secs (59.26 usr +  0.17 sys = 59.43 CPU)
+    50: 445 wallclock secs (47.32 usr +  0.21 sys = 47.53 CPU)
+
+=head2 SearchDirectoryRequest limit
+
+The ZCS SearchDirectoryRequest SOAP call provides paged results.
+Below are timings with various values for 'limit' with 'maxResults'
+set to zero (0).
+
+Test directory with more than 83000 entries.  A limit of 50 was chosen
+after these tests.
+
+ limit: timing via time(1)
+ ------ --------------------
+   500: real 21m19.636s user 20m42.027s sys 0m0.395s
+   200: real  9m36.827s user  8m52.900s sys 0m0.717s
+   100: real  5m47.554s user  4m50.981s sys 0m0.769s
+    75: real  4m51.761s user  3m54.414s sys 0m0.799s
+    50: real  4m17.056s user  3m02.626s sys 0m1.246s
+    40: real  4m22.345s user  2m57.274s sys 0m1.530s
+
+=head1 BUGS/KNOWN ISSUES
+
+=over 3
```

Comment:
nit: change this to '4' from '3' (an indentation of 3 is unusual in POD)

Reply:
done